packages/mgt/src/graph/graph.groups.ts

```diff
@@ -64,6 +97,16 @@ export async function findGroups(
 ): Promise<Group[]> {
   const scopes = 'Group.Read.All';
 
+  let cache: CacheStore<CacheGroups>;
+
+  if (groupsCacheEnabled()) {
+    cache = CacheService.getCache(cacheSchema, 'groups');
+    const groups = await cache.getValue(query);
```

Comment:
You are only storying the query, but the results can be different based on the `top` and `groupTypes` parameters. You should be caching these params as well and deciding based on these whether the cached value is appropriate.